### ISSUE TITLE
Reduce the amount of stateful behaviour in genPGM.py

### ISSUE THREAD

**Issue by adarshp** [ml4ai/delphi#83]
I'm having some issues with global state in `genPGM.py` - basically, the `lambdas.py` file seems to change upon multiple runs of the same function with the same inputs.

Steps to reproduce (assuming you are in the Delphi repo root directory) - 

```bash
cd delphi/program_analysis/autoTranslate
./autoTranslate ../../../data/program_analysis/crop_yield.f
python
```

Then in the Python interpreter, do:

```python
>>> from delphi.program_analysis.autoTranslate.scripts.genPGM.import get_asts_from_files, create_pgm_dict
>>> asts = get_asts_from_files(['crop_yield.py'])
>>> pgm_dict = create_pgm_dict('lambdas.py', asts, 'pgm.json')
```

The `lambdas.py` file is unchanged by this.

```python
def UPDATE_EST__lambda__TOTAL_RAIN_0(TOTAL_RAIN, RAIN):
    TOTAL_RAIN = (TOTAL_RAIN+RAIN)
    return TOTAL_RAIN

def UPDATE_EST__lambda__IF_1_0(TOTAL_RAIN):
    return (TOTAL_RAIN<=40)

def UPDATE_EST__lambda__YIELD_EST_0(TOTAL_RAIN):
    YIELD_EST = (-((((TOTAL_RAIN-40)**2)/16))+100)
    return YIELD_EST

def UPDATE_EST__lambda__YIELD_EST_1(TOTAL_RAIN):
    YIELD_EST = (-(TOTAL_RAIN)+140)
    return YIELD_EST

def CROP_YIELD__lambda__MAX_RAIN_0():
    MAX_RAIN = 4.0
    return MAX_RAIN

def CROP_YIELD__lambda__CONSISTENCY_0():
    CONSISTENCY = 64.0
    return CONSISTENCY

def CROP_YIELD__lambda__ABSORPTION_0():
    ABSORPTION = 0.6
    return ABSORPTION

def CROP_YIELD__lambda__YIELD_EST_0():
    YIELD_EST = 0
    return YIELD_EST

def CROP_YIELD__lambda__TOTAL_RAIN_0():
    TOTAL_RAIN = 0
    return TOTAL_RAIN

def CROP_YIELD__lambda__RAIN_0(DAY, CONSISTENCY, MAX_RAIN, ABSORPTION):
    RAIN = ((-((((DAY-16)**2)/CONSISTENCY))+MAX_RAIN)*ABSORPTION)
    return RAIN
```

However, upon calling this function a second time:

```python
>>> pgm_dict = create_pgm_dict('lambdas.py', asts, 'pgm.json')
```

The numbers at the end of the function names in `lambdas.py` get incremented by one. 

```python
def UPDATE_EST__lambda__TOTAL_RAIN_1(TOTAL_RAIN, RAIN):
    TOTAL_RAIN = (TOTAL_RAIN+RAIN)
    return TOTAL_RAIN

def UPDATE_EST__lambda__IF_1_1(TOTAL_RAIN):
    return (TOTAL_RAIN<=40)

def UPDATE_EST__lambda__YIELD_EST_2(TOTAL_RAIN):
    YIELD_EST = (-((((TOTAL_RAIN-40)**2)/16))+100)
    return YIELD_EST

def UPDATE_EST__lambda__YIELD_EST_3(TOTAL_RAIN):
    YIELD_EST = (-(TOTAL_RAIN)+140)
    return YIELD_EST

def CROP_YIELD__lambda__MAX_RAIN_1():
    MAX_RAIN = 4.0
    return MAX_RAIN

def CROP_YIELD__lambda__CONSISTENCY_1():
    CONSISTENCY = 64.0
    return CONSISTENCY

def CROP_YIELD__lambda__ABSORPTION_1():
    ABSORPTION = 0.6
    return ABSORPTION

def CROP_YIELD__lambda__YIELD_EST_1():
    YIELD_EST = 0
    return YIELD_EST

def CROP_YIELD__lambda__TOTAL_RAIN_1():
    TOTAL_RAIN = 0
    return TOTAL_RAIN

def CROP_YIELD__lambda__RAIN_1(DAY, CONSISTENCY, MAX_RAIN, ABSORPTION):
    RAIN = ((-((((DAY-16)**2)/CONSISTENCY))+MAX_RAIN)*ABSORPTION)
    return RAIN
```

This side effect needs to be gotten rid of.

**Comment by pauldhein** [comment]
Found a global variable that was being used to track indexes for function names. Removed the use of the global variable in favor of another argument to `genPGM()`. This issue is now fixed and the fix is present in 10cd2d7.

**Comment by adarshp** [comment]
Excellent, thanks @pauldhein !
